sdks/python/apache_beam/runners/interactive/display/pcoll_visualization.py

```diff
@@ -0,0 +1,279 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Module visualizes PCollection data.
+
+For internal use only; no backwards-compatibility guarantees.
+Only works with Python 3.5+.
+"""
+from __future__ import absolute_import
+
+import base64
+import logging
+from datetime import timedelta
+
+from pandas.io.json import json_normalize
+
+from apache_beam import pvalue
+from apache_beam.runners.interactive import interactive_environment as ie
+from apache_beam.runners.interactive import pipeline_instrument as instr
+
+# jsons doesn't support < Python 3.5. Work around with json for legacy tests.
+# TODO(BEAM-8288): clean up once Py2 is deprecated from Beam.
+try:
+  import jsons
+  _pv_jsons_load = jsons.load
+  _pv_jsons_dump = jsons.dump
+except ImportError:
+  import json
+  _pv_jsons_load = json.load
+  _pv_jsons_dump = json.dump
+
+try:
+  from facets_overview.generic_feature_statistics_generator import GenericFeatureStatisticsGenerator
+  _facets_gfsg_ready = True
+except ImportError:
+  _facets_gfsg_ready = False
```

Comment:
Is there a reason for _facets_gfsg_ready, _ipython_ready, _tl_ready to be separately ready? Does this module work with some of them installed and some not?

I think this import section could be reduced to 

try:
... import all
except ImportError
... warning, missing dependencies and ask user to install [interactive]

Reply:
They can work separately.
If not _ipython_ready, the user is not even in an interactive environment. There is no kernel, so there is definitely no frontend connected to the kernel. Thus no notebook, nowhere to visualize, so basically everything is NOOP.
If not _facets_gfsg_ready, the user wouldn't see the facets-dive widget, but they can see facets-overview and data-table widgets.
If not _tl_ready, the user wouldn't see dynamic plotting but one-shot plotting.

Warning about the user to install [interactive] is a great suggestion!

I can 
1. make it import all and NOOP if any of the dependency is not ready
2. make it import separately and try-best to do visualization
Either way, I'll add the warning for users to `pip install apache-beam[interactive]`. Which option do you like?

Reply:
I would suggest first option. Require installation of full [interactive] extra package.